runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -284,11 +284,15 @@ public IClientRouter addClient(IClient client) {
         client.setRouter(this);
 
         // Iterate through all types of CorfuMsgType, registering the handler
-        client.getHandledTypes()
-                .forEach(x -> {
-                    handlerMap.put(x, client);
-                    log.trace("Registered {} to handle messages of type {}", client, x);
-                });
+        try {
+            client.getHandledTypes()
+                    .forEach(x -> {
+                        handlerMap.put(x, client);
+                        log.trace("Registered {} to handle messages of type {}", client, x);
+                    });
+        } catch (Exception ex) {
+            // No Legacy getHandledTypes - Ignore
+        }
```

Comment:
instead catching all exceptions can you just ignore unsupported operation and log it?

Also, why was this not an issue when we pushed in the new BaseServer changes?

Reply:
Will log and catch `UnsupportedOperation`. It was an issue for the `BaseServer`, but not for the `BaseHandler` because there were still some old `CorfuMsg` types that needed to be handled by it. Since it wasn't removed, no exception was thrown.

Reply:
Fixed. Only catching `UnsupportedOperation` and logging accordingly.